src/main/java/hlf/java/rest/client/service/impl/NetworkStatusImpl.java

```diff
@@ -293,17 +327,69 @@ public ResponseEntity<ClientResponseModel> addOrgToChannel(
     if (network != null && user != null) {
       try {
         Channel selectedChannel = network.getChannel();
-        ConfigUpdate selectedChannelConfigUpdate =
-            ConfigUpdate.parseFrom(selectedChannel.getChannelConfigurationBytes());
-        // ReadSet must have version, all other fields ignored; WriteSet must have changes
-        ConfigGroup readset = selectedChannelConfigUpdate.getReadSet();
-        // ConfigGroups consist of: groups, modPolicy, policies, values, and version.
+        ConfigUpdate.Builder configUpdateBuilder = createConfigUpdate(channelName);
+        ConfigUpdate configUpdate =
+            configUpdateBuilder
+                .setWriteSet(
+                    addOrgToChannelWriteSetBuilder.buildWriteset(
+                        configUpdateBuilder.getReadSet(), organizationDetails))
+                .build();
+        MessageOrBuilder message = configUpdate;
+        String channelConfigString = JsonFormat.printer().print(message);
```

Comment:
a method could be extracted for fetching channel config to be reused in generateConfigUpdate method as well

Reply:
Sure, addressed.